diff --git a/hw/riscv/ws63.c b/hw/riscv/ws63.c
index 7ce56d1..eab1eb4 100644
--- a/hw/riscv/ws63.c
+++ b/hw/riscv/ws63.c
@@ -44,9 +44,9 @@
 #include "chardev/char-fe.h"
 #include "target/riscv/cpu.h"
 #include "hw/core/cpu.h"
-#include "system/system.h"
-#include "system/reset.h"
-#include "system/runstate.h"
+#include "sysemu/sysemu.h"
+#include "sysemu/reset.h"
+#include "sysemu/runstate.h"
 #include "exec/address-spaces.h"
 #include "elf.h"
 #include "trace.h"     /* generated from hw/riscv/trace-events (ws63_* events) */
@@ -421,8 +421,9 @@ static void ws63_uart_realize(DeviceState *dev, Error **errp)
                              ws63_uart_event, NULL, s, NULL, true);
 }
 
-static const Property ws63_uart_properties[] = {
+static Property ws63_uart_properties[] = {
     DEFINE_PROP_CHR("chardev", WS63UartState, chr),
+    DEFINE_PROP_END_OF_LIST(),
 };
 
 static void ws63_uart_instance_init(Object *obj)
@@ -1792,8 +1793,9 @@ static void ws63_netmac_realize(DeviceState *dev, Error **errp)
     qemu_format_nic_info_str(qemu_get_queue(s->nic), s->conf.macaddr.a);
 }
 
-static const Property ws63_netmac_props[] = {
+static Property ws63_netmac_props[] = {
     DEFINE_NIC_PROPERTIES(WS63NetMacState, conf),
+    DEFINE_PROP_END_OF_LIST(),
 };
 
 static void ws63_netmac_class_init(ObjectClass *klass, void *data)
